memory-db/src/malloc_size_of.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module contains traits and structs related to the `MallocSizeOf` trait.
+
+use core::marker::PhantomData;
+use parity_util_mem::{malloc_size, MallocSizeOf};
+
+
+/// Used to implement incremental evaluation of `MallocSizeOf` for a collection.
+pub trait MallocSizeOfCallback<T> {
+	/// Update `malloc_size_of` when a value is removed.
+	fn on_value_removed(&mut self, _value: &T) {}
+	/// Update `malloc_size_of` when a value is inserted.
+	fn on_value_inserted(&mut self, _value: &T) {}
```

Comment:
```suggestion
	fn on_insert(&mut self, _value: &T) {}
```
My thinking here is that "value" doesn't really add much info (it's always a value being removed/inserted?).